shopfloor/services/delivery.py

```diff
@@ -312,6 +312,11 @@ def _deliver_product(self, picking, product, product_qty=None, location=None):
             limit=limit,
             order="date_planned",
         )
+        # Filter lines to keep only ones from one delivery operation
+        # (we do not want to process lines of another delivery operation)
+        if len(lines.picking_id) > 1:
+            lines_picking = fields.first(picking or lines.picking_id)
+            lines = lines.filtered_domain([("picking_id", "=", lines_picking.id)])
```

Comment:
This should then be placed after the multiple_location check.

Reply:
And after the package check

Reply:
Done, I also move this code in a dedicated method on `stock.move.line`.